### PR TITLE
Changed Help and Feedback Activity to Fragment

### LINKED ISSUE #770: Help & feedback layout not having toolbar
**Actual Behavior**
Just white void layout comes up.

**Expected Behavior**
Should have FAQ and help instructions.

**Steps to reproduce it**
Simply implementing toolbar layout and back button functionality on it. 

**Screen-shot of the issue**
bug_report

**Would you like to work on the issue?**

_**Yes**_,I would like to work on this.

### PR DESCRIPTION
**Pull request by harsh-2711**
Fixes #770 

Changes: Changed Help and Feedback Activity to fragment and provided navigation drawer with it.

Screenshots for the change: 
screenshot_20180517-201226

APK for testing: 
[app-debug.zip](https://github.com/fossasia/pslab-android/files/2013632/app-debug.zip)

### CODE REVIEW

File: app/src/main/java/org/fossasia/pslab/fragment/HelpAndFeedbackFragment.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+    public static HelpAndFeedbackFragment newInstance() {
+        return new HelpAndFeedbackFragment();
+    }
+
+    @BindView(R.id.appBarAnim)
+    AppBarLayout appBarLayout;
+
+    public HelpAndFeedbackFragment() {
+        // Required empty public constructor
```
Comment (CloudyPadmal) [inline app/src/main/java/org/fossasia/pslab/fragment/HelpAndFeedbackFragment.java]: These kind of comments are not needed

File: app/src/main/java/org/fossasia/pslab/fragment/HelpAndFeedbackFragment.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    }
+
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        // Inflate the layout for this fragment
```
Comment (CloudyPadmal) [inline app/src/main/java/org/fossasia/pslab/fragment/HelpAndFeedbackFragment.java]: This one also please remove

File: app/src/main/java/org/fossasia/pslab/fragment/HelpAndFeedbackFragment.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        View view = inflater.inflate(R.layout.fragment_help_feedback, container, false);
+        ButterKnife.bind(this, view);
+
+        return view;
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        // handle arrow click here
```
Comment (CloudyPadmal) [inline app/src/main/java/org/fossasia/pslab/fragment/HelpAndFeedbackFragment.java]: This one too

File: app/src/main/java/org/fossasia/pslab/activity/MainActivity.java [hunk 4]
```diff
@@ -185,7 +189,10 @@ private void setToolbarTitle() {
     }
 
     private void selectNavMenu() {
-        navigationView.getMenu().getItem(navItemIndex).setChecked(true);
+        if(navItemIndex == 4)
+            navigationView.getMenu().getItem(3).setChecked(true);
```
Comment (CloudyPadmal) [inline app/src/main/java/org/fossasia/pslab/activity/MainActivity.java]: Can you please adjust array entries not to have an `if` check here?

**Comment by harsh-2711** [comment]
@CloudyPadmal  I have removed the comments but I can't figure out how to adjust array entries as there are three initial entries and one last common entry which includes sub array with three entries. So, if navItemIndex goes beyond three than it will give error of ArrayOutOfBound

**Comment by CloudyPadmal** [comment]
Have you actually tested if `navigationView.getMenu().getItem(3).setChecked(true);` will highlight "Help & Feedback" in the navigation menu? Shouldn't it be `navigationView.getMenu().getItem(3).getSubMenu().getItem(0).setChecked(true);`?

**Comment by harsh-2711** [comment]
@CloudyPadmal  That's true.. I tried that too but results were same and we need if statements for both the cases. I can make that changes but we need one if..else statement. So, should I proceed to make suggested changes?

**Comment by CloudyPadmal** [comment]
Yes @harsh-2711 ! We can add that if statement!

**Comment by harsh-2711** [comment]
@CloudyPadmal  Changes done as requested

**Comment by CloudyPadmal** [comment]
Thank you @harsh-2711 ! PR looks good. Can you generate apk file and provide a link here to test?

**Comment by harsh-2711** [comment]
[app-debug.zip](https://github.com/fossasia/pslab-android/files/2019314/app-debug.zip)

**Comment by CloudyPadmal** [comment]
Hello @harsh-2711 ! When I click on `Help & Feedback` now the menu item highlights. But it also keeps on highlighting `Instruments` or the previous place we were in. Can you make those unchecked when moved to `Help & Feedback`?

**Comment by harsh-2711** [comment]
@CloudyPadmal  I made the necessary changes for both Help and Feedback Fragment and also for About Us as it was not highlighting before. Also I have used switch case to handle the exception that occurs when any new fragment is not selected but is tried to open (I faced this problem). Here is the apk to test [app-debug.zip](https://github.com/fossasia/pslab-android/files/2020319/app-debug.zip)

File: app/src/main/java/org/fossasia/pslab/activity/MainActivity.java [hunk 5]
```diff
@@ -185,7 +189,32 @@ private void setToolbarTitle() {
...
+            case 4:
+                navigationView.getMenu().getItem(3).getSubMenu().getItem(0).setChecked(true);
+                navigationView.getMenu().getItem(3).getSubMenu().getItem(1).setChecked(false);
+                navigationView.getMenu().getItem(0).setChecked(false);
+                navigationView.getMenu().getItem(1).setChecked(false);
+                navigationView.getMenu().getItem(2).setChecked(false);
+                break;
+            default:
+                navigationView.getMenu().getItem(0).setChecked(true);
+                navigationView.getMenu().getItem(3).getSubMenu().getItem(1).setChecked(false);
```
Comment (CloudyPadmal) [inline app/src/main/java/org/fossasia/pslab/activity/MainActivity.java]: This part looks very big. Let me see if there's a simpler way of doing it.

**Comment by harsh-2711** [comment]
@CloudyPadmal  Code optimized... Thanks for help

**Comment by CloudyPadmal** [comment]
@harsh-2711 Travis is failing. It's a code error. Please try to run the code in your phone when you make a code change before pushing.

**Comment by harsh-2711** [comment]
Sorry @CloudyPadmal but I ran the code on my phone and it worked completely. But mistake was I mistakenly merged my other branch with this branch. So I will revert the commit and will than push the changes.

**Comment by harsh-2711** [comment]
@CloudyPadmal  Please review it now..

File: app/src/main/java/org/fossasia/pslab/fragment/HelpAndFeedbackFragment.java [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+ */
+
+public class HelpAndFeedbackFragment extends Fragment {
+
+    public static HelpAndFeedbackFragment newInstance() {
+        return new HelpAndFeedbackFragment();
+    }
+
+    @BindView(R.id.appBarAnim)
+    AppBarLayout appBarLayout;
```
Comment (CloudyPadmal) [inline app/src/main/java/org/fossasia/pslab/fragment/HelpAndFeedbackFragment.java]: In Java we normally define variables at the beginning of class. line `27`, `28` should come to `22`. Please make the changes in @abhinavraj23 's similar implementation of About Us fragment as well. It seems like you've just copy pasted his code.

**Comment by harsh-2711** [comment]
Sorry for ill format. Will take care from now. Changes done. Please review them

**Pull request merged**